src/pybind/mgr/mgr_module.py

```diff
@@ -17,39 +17,39 @@
 
 # Full list of strings in "osd_types.cc:pg_state_string()"
 PG_STATES = [
+    "activating",
```

Comment:
please avoid blending the cosmetic changes with functional changes. in your case, i cannot identify what is actually added. furthermore to "sync" the list of string representation of "PG_STATES", please keep the order of the strings here also in sync with them in `pg_state_string()`, instead of order them in alphabetical order. otherwise, it'd be more difficult to add/remove states if they are out of sync again in future.

Reply:
also, i think a more future-proof fix for https://tracker.ceph.com/issues/46142 is to use `collections.defaultdict` to aggregate the number of PGs in different states.

Reply:
> please avoid blending the cosmetic changes with functional changes. in your case, i cannot identify what is actually added. furthermore to "sync" the list of string representation of "PG_STATES", please keep the order of the strings here also in sync with them in `pg_state_string()`, instead of order them in alphabetical order. otherwise, it'd be more difficult to add/remove states if they are out of sync again in future.

It was not just cosmetic. If the state is not within the PG_STATES list, then no pg_%STATE% metric created in prometheus module. E.g. pg_stale. Ok, I will reorder the list to make as few changes as possible.


> also, i think a more future-proof fix for https://tracker.ceph.com/issues/46142 is to use `collections.defaultdict` to aggregate the number of PGs in different states.

Yes, definitely. Will return back shortly.

Reply:
My bad. "laggy" was already there and PG_STATE was in sync. It was not in the PG_STATE  in 15.2.3 from the issue. I really should mix these two commits.
Ok, now here is only one commit with defaultdict. defaultdict allows avoiding situations like in https://tracker.ceph.com/issues/46142 when PG_STATE is not in sync with pg_state_string()